src/modules/service-worker-remover/index.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  VERSIONED_CACHE_NAME,
+  UNVERSIONED_CACHE_NAME,
+} from '../amp-caching/constants';
+import { cacheName as AMP_ASSET_CACHE } from '../asset-caching/constants';
+import { cacheName as AMP_PUBLISHER_CACHE } from '../document-caching/constants';
+import { cacheName as AMP_PREFETCHED_LINKS } from '../link-prefetch/constants';
+
+export class ServiceWorkerRemover {
+  async installNoOpServiceWorker() {
```

Comment:
Let's verify the first version of Safari supporting Service Workers, there is a bug in Safari 10.1 that would preclude an async member from working as expected.

Reply:
11.2